packages/odyssey-react-mui/src/Autocomplete.tsx

```diff
@@ -286,287 +255,35 @@ const Autocomplete = <
-    HTMLDivElement,
-    HTMLAttributes<HTMLElement>
-  >(function (props, ref) {
-    const [listHeight, setListHeight] = useState(0);
-
-    const { children, ...other } = props;
-    const itemData: ReactElement[] = (children as ReactElement[]).flatMap(
-      (item: ReactElement & { children?: ReactElement[] }) =>
-        [item].concat(item.children || []),
-    );
-
-    const sizeMap = useRef<number[]>([]);
-
-    const getListBoxHeight = useCallback(() => {
-      // 8px of padding top/bottom applied by MUI
-      const COMBINED_LISTBOX_PADDING = 16;
-
-      if (itemData.length > OVERSCAN_ROW_COUNT) {
-        // has a max-height of 40vh set in CSS. This is only set because height needs to be a number
-        return 99999;
-      } else {
-        const itemsHeightCalculated = itemData
-          .map((_, index) => sizeMap.current[index] || 0)
-          .reduce(
-            (prevItemHeight, nextItemHeight) => prevItemHeight + nextItemHeight,
-            0,
-          );
-        return COMBINED_LISTBOX_PADDING + itemsHeightCalculated;
-      }
-    }, [itemData, sizeMap]);
-
-    useEffect(() => {
-      if (sizeMap.current.length && itemData.length) {
-        setListHeight(getListBoxHeight());
-      }
-    }, [getListBoxHeight, itemData, sizeMap]);
-
-    // The number of items (rows or columns) to render outside of the visible area for performance and scrolling reasons
-    const OVERSCAN_ROW_COUNT = 8;
-
-    const gridRef = useResetCache(itemData.length);
-
-    const setItemSize = useCallback<SetItemSize>(
-      (size) => {
-        gridRef?.current?.resetAfterIndex(0, true);
-        sizeMap.current = sizeMap.current.concat(size);
-      },
-      [gridRef, sizeMap],
-    );
-    const getItemSize = useCallback(
-      // using 45px as a sane default here to avoid a lot of content shift on repaint
-      (index: number) => sizeMap.current[index] || 45,
-      [sizeMap],
-    );
-
-    return (
-      <ListboxContainer ref={ref}>
-        <OuterListboxContext.Provider value={other}>
-          <VariableSizeList
-            innerElementType="ul"
-            itemData={itemData}
-            itemCount={itemData.length}
-            itemSize={getItemSize}
-            height={listHeight}
-            width="100%"
-            ref={gridRef}
-            outerElementType={OuterListboxElementType}
-            overscanCount={OVERSCAN_ROW_COUNT}
-          >
-            {({ data, index, style }) => (
-              <Row
-                data={data}
-                index={index}
-                style={style}
-                setItemSize={setItemSize}
-              />
-            )}
-          </VariableSizeList>
-        </OuterListboxContext.Provider>
-      </ListboxContainer>
-    );
+  const {
+    inputValueProp,
+    isVirtualized,
+    onChange,
+    onInputChange,
+    renderInput,
+    t,
+    valueProps,
+    ListboxComponent,
+  } = useAutocomplete<OptionType, HasMultipleChoices, IsCustomValueAllowed>({
```

Comment:
This shouldn't return `t` right? If we need the translation function, we should grab it from that `useTranslation` hook right?

Reply:
@KevinGhadyani-Okta I had mixed feelings about this. Felt a little weird to return it here but, I also felt like it might make sense to do it to keep from calling the hook again. I'm good either way